### ISSUE TITLE
SageAdapter: No accession found

### ISSUE THREAD

**Issue by elrubio** [OpenMS/OpenMS#7112]
`Version: 3.1.0-pre-nightly-2023-10-05 Oct  5 2023, 23:04:00, Revision: b18b52e`

When running `SageAdapter` with `-reindex false` (and only then), the following error occurs: 

> SageAdapter -ini /data/oms_sage_adapter.ini -in /data/file.mzML -out /data/file.idXML -database /data/UP000000589_10090.decoy.fasta **-reindex false**
> Found Sage version string: Version 0.14.1
> Creating temp file name...
> Creating Sage config file.../tmp/20231006_083047_74e504abe0bf_1_1.json
> Sage command line: /thirdparty/Sage/sage /tmp/20231006_083047_74e504abe0bf_1_1.json -f /data/UP000000589_10090.decoy.fasta -o /data --write-pin /data/file.mzML
> Reading sage output...
> **Error: Unexpected internal error (the element 'No accession sp|P62880|GBB2_MOUSE found in run 'Sage_2023-10-06 08:32:04' for PSM LLLAGYDDFNC(Carbamidomethyl)NIWDAMK_2. Please contact the maintainer of this tool e.g. on GitHub as this should not happen.' could not be found)**
> 

Notes:
- Without  `-reindex false`, everything runs fine
- The fasta file was created with `DecoyDatabase`, using default parameters
- I can provide the files if requested

**Comment by jpfeuffer** [comment]
I don't think we currently support parsing proteins from PIN files, so this either needs to be added by also returning protein IDs here (https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/PercolatorInfile.cpp#L59) or reindexing needs to be mandatory in Adapters that are loading PIN files.
